projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.ts

```diff
@@ -134,13 +135,20 @@ const getVideoValidationError = ( sizeInMb, length, width, height, videoLimits )
  * @param {object} mediaData - Data for media, width, height, source_url etc.
  * @param {string} serviceName - The name of the social media service we want to validate against. facebook, tumblr etc.
  * @param {boolean} hasAttachedMedia - Whether the media is attached.
+ * @param {boolean} hasAutoGeneratedImage - Whether there is an auto generated image.
  * @returns {(FILE_SIZE_ERROR | FILE_TYPE_ERROR | VIDEO_LENGTH_TOO_SHORT_ERROR | VIDEO_LENGTH_TOO_LONG_ERROR)} Returns validation error.
  */
-const getValidationError = ( metaData, mediaData, serviceName, hasAttachedMedia ) => {
+const getValidationError = (
+	metaData,
+	mediaData,
+	serviceName,
+	hasAttachedMedia,
+	hasAutoGeneratedImage
+) => {
 	const restrictions = RESTRICTIONS[ serviceName ] ?? DEFAULT_RESTRICTIONS;
 
 	if ( ! metaData || Object.keys( metaData ).length === 0 ) {
-		return restrictions.requiresMedia ? NO_MEDIA_ERROR : null;
+		return restrictions.requiresMedia && ! hasAutoGeneratedImage ? NO_MEDIA_ERROR : null;
```

Comment:
Shouldn't this be `return restrictions.requiresMedia && ( ! hasAutoGeneratedImage && serviceName === 'instagram-business'  ) ? NO_MEDIA_ERROR : null;` 

Did you have any reason behind naming the variable `hasAutoGeneratedImage` instead of `hasSIGImage`?